rfcs/0114-nixos-code-of-conduct.md

```diff
@@ -0,0 +1,108 @@
+---
+feature: NixOS Code of Conduct
+start-date: 2021-11-03
+author: @jonringer
+co-authors:
+shepherd-team:
+shepherd-leader:
+related-issues: #98 #102
+---
+
+# Summary
+[summary]: #summary
+
+The NixOS Community needs some foundation of shared values which can be used
+to determine appropriate when contributing and interacting with the community.
+We currently have a [community ethos](https://nixos.org/community/index.html), but
+this only states that we are an inclusive community, but doesn't define
+inappropriate behavior when interacting with the community. This RFC aims
+to establish the shared values, so they can be used to identify behavior
+which is disruptive to the community.
+
+This RFC does not intend to define moderation practices. This is intended to only lay
+a framework of shared values for the community.
+
+# Motivation
+[motivation]: #motivation
+
+The NixOS community struggles with a clearly defined way in which someone's behavior may
+be disruptive. A well-defined Code of Conduct gives an explicit
+set of expectations for all contributors, which will also make it easier to identify
+when behavior becomes disruptive.
+
+# Detailed design
+[design]: #detailed-design
+
+These statutes are taken from [rust-lang's Code of Conduct](https://www.rust-lang.org/policies/code-of-conduct),
+which provides a great compromise between explicit behaviors and subjective goals.
+Only the moderation body name has been changed.
+
+- We are committed to providing a friendly, safe and welcoming environment for
+all, regardless of level of experience, gender identity and expression,
+sexual orientation, disability, personal appearance, body size, race, ethnicity, age,
+religion, nationality, or other similar characteristics.
+- Please avoid using overtly sexual aliases or other nicknames that might
+detract from a friendly, safe and welcoming environment for all.
+- Please be kind and courteous. There’s no need to be mean or rude.
+- Respect that people have differences of opinion and that every design or
+implementation choice carries a trade-off and numerous costs. There is seldom a right answer.
```

Comment:
```suggestion
implementation choice carries a trade-off and numerous costs. There is seldom a single right answer.
```

Reply:
Also, I would maybe put this bullet point higher. This point might be a contributory factor to most GitHub bans…